project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -148,8 +189,137 @@ record ProjectContact(String fullName, String email) {
+   * @since 1.9.0
+   */
+  record ExperimentalVariable(String name, Set<String> levels, @Nullable String unit) {
+
+  }
+
+  /**
+   * Container of experimental variables. Can be used in {@link #update(ExperimentUpdateRequest)}.
+   * @param experimentalVariables the list of experimental variables
+   * @since 1.9.0
+   */
+  record ExperimentalVariables(
+      List<ExperimentalVariable> experimentalVariables) implements
+      ExperimentUpdateRequestBody,
+      ExperimentUpdateResponseBody {
+
+  }
+
+  /**
+   * A level of an experimental variable
+   *
+   * @param variableName the name of the variable
+   * @param levelValue   the value of the level
+   * @param unit         the unit for the value of the level. Can be null if no unit is set
+   * @since 1.9.0
+   */
+  record VariableLevel(String variableName, String levelValue, @Nullable String unit) {
+
+  }
+
+  /**
+   * Information about an experimental group
+   *
+   * @param groupId    the identifier of the group
+   * @param name       the name of the eperimental group can be empty but is not expected to be
+   *                   null
+   * @param sampleSize the number of samples in this experimental group
+   * @param levels     the experimental variable levels making up the condition for the samples in
+   *                   this group.
+   * @since 1.9.0
+   */
+  record ExperimentalGroup(@Nullable Long groupId, String name, int sampleSize,
+                           Set<VariableLevel> levels) {
+
+  }
+
+  /**
+   * A container for experimental groups. Can be used in {@link #update(ExperimentUpdateRequest)}
+   * @param experimentalGroups the list of experimental groups
+   * @since 1.9.0
+   */
+  record ExperimentalGroups(List<ExperimentalGroup> experimentalGroups) implements
+      ExperimentUpdateRequestBody,
+      ExperimentUpdateResponseBody {
+
+  }
+
+  /**
+   * A container describing the experiment
+   *
+   * @param experimentName the name of the experiment
+   * @param species        a set of species for the experiment. Expected textual representations
+   *                       containing CURIEs.
+   * @param specimen       a set of specimen for the eperiment. Expected textual representations
+   *                       containing CURIEs.
+   * @param analytes       a set of analytes for the eperiment.Expected textual representations
+   *                       containing CURIEs.
+   * @since 1.9.0
+   */
+  record ExperimentDescription(String experimentName, Set<String> species, Set<String> specimen,
+                               Set<String> analytes) implements ExperimentUpdateRequestBody,
+      ExperimentUpdateResponseBody {
+
+
+  }
+
+  /**
+   * A list of confounding variable information. Can be used in {@link #update(ExperimentUpdateRequest)}
+   * @param confoundingVariables the variable information
+   */
+  record ConfoundingVariables(List<ConfoundingVariableInformation> confoundingVariables) implements
```

Comment:
I am not sure if immutability is guaranteed with that approach. Can you comment on that?

Reply:
yes, just confirmed it. We should ensure and promote immutability of its parameters, so I favor sth like:

```java
record ListRecord(List<String> content){
    public ListRecord{
      content = List.copyOf(content);
    }
  }
```

Reply:
to ensure side-effect free records :)

Reply:
Or we could directly ask for an UnmodifiableList